### PR TITLE
Calendly V2 Api Node

### PR DESCRIPTION
**Pull request by tcampb**
### What is the purpose of this PR?

This PR addresses [issue 3392](https://github.com/n8n-io/n8n/issues/3392):

> Calendly is phasing out support for v1 API keys June 30th 2022. The current Calendly Node only authenticates using v1 API keys. Trying to use Calendly tokens does not work.

### Changes
* Updates the current Calendly node display names to include `Legacy`; this terminology is used by Calendly to describe their [V1 api](https://developer.calendly.com/api-docs/YXBpOjI3ODE4MDY3-legacy-calendly-api-v1).
* Adds `calendlyV2Trigger` node type and `calendlyV2Api` credentials.

The majority of this code was copied from the current Calendly node implementation with a few changes added to the webhook methods to make it compatible with the v2 api (see code comments for additional details).

image

### Test Cases

1. Create `User` Scope Webhook using Organization Owner Personal Access Token
https://monosnap.com/file/Jun9nvN56aeYOmufWc6chONCl7Iun6

2. Create `Organization` Scope Webhook using Organization Owner Personal Access Token
https://monosnap.com/file/YGMEPFuG9Nh7PKqFwGYZhPIG567MOQ

3. Create `User` Scope Webhook using Personal Access Token **not** associated with the Organization Owner / Admin
https://monosnap.com/file/zZ5KuYMHF8LNrYD36xMobkyf1JdcVd

### Documentation Updates
https://github.com/n8n-io/n8n-docs/pull/876

### CODE REVIEW

File: packages/nodes-base/nodes/CalendlyV2/CalendlyTrigger.node.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+
+				const { resource } = await calendlyApiRequest.call(this, 'GET', '/users/me', {});
+
+				// Check all the webhooks which exist already if it is identical to the
+				// one that is supposed to get created.
+				const endpoint = '/webhook_subscriptions';
+				const query = {
+					scope,
+					organization: resource.current_organization,
+					count: 100,
```
Comment (tcampb) [inline packages/nodes-base/nodes/CalendlyV2/CalendlyTrigger.node.ts:124]: I'm not sure what the best practice is here for determining if a webhook already exists. The current [webhook subscriptions api ](https://developer.calendly.com/api-docs/faac832d7c57d-list-webhook-subscriptions)does not allow you to search by url or the webhook state. 

There could be many webhooks for the Calendly organization that are disabled so I didn't want to send too many requests here. For now I just check the first 100 webhook subscriptions that are returned.

**Comment by cheap-geek** [comment]
Is there a way for me to download and try this version fix for Calendly?

**Comment by Joffcom** [comment]
Hey @tcampb,

I was just looking at this one, From what I can see our current node supports both the v1 and v2 api. Does this add anything else that the current node doesn't include?

**Comment by tcampb** [comment]
Hey @Joffcom, it does not, I originally created this before the Calendly node supported v2. Will close this PR, thanks for the reminder!

**Pull request closed without merging**